testpar/t_coll_md_read.c

```diff
@@ -34,10 +34,9 @@
 
```

Comment:
All the changes in this file were just to make one test more reproducible. It used to be that the test would only reliably reproduce the issue on exactly 2 or 3 MPI ranks (see Condition 3 from the old code below), but I refactored the test a bit so it should be reproducible for any number of MPI ranks.